blog/2017-08-10-RealTimePlanningUsingKieServerPart1.adoc

```diff
@@ -0,0 +1,149 @@
+= Optimize your problems using KIE Server, part 1
+:page-interpolate: true
+:awestruct-author: mcimbora
+:awestruct-layout: blogPostBase
+:awestruct-tags: [feature, howto, kie server]
+
+KIE Server provides an execution environment separated from your client application.
+Run it on the same local network as your client, or in the cloud, depending on your needs.
+The server capabilities are exposed via JMS/REST interface, with the latter making the content easily consumable
+by non-Java clients.
+This is extremely useful if you want to leverage OptaPlanner outside Java ecosystem.
+In the 7 release stream, we worked to improve OptaPlanner/KIE Server integration.
+Apart from API improvements I am happy to announce the server now supports real-time planning.
+Stay tuned for part two.
+
+== New OptaPlanner REST API overview
+
+The following table shows an overview of the new API introduced in version 7.0.0.Final.
+All solver action URLs are normally prefixed by container URL, e.g. `http://${kie-server}/services/rest/server/containers/${container_id}`,
+which is not included in the table to keep the output short.
+
+|===
+|Action |Method |URL suffix | Payload
+
+|Register solver a|PUT a|solvers/${solver_id} |-
+|Get solver a|GET a|solvers/${solver_id} |SolverInstance object
+|Submit solution a|POST a|solvers/${solver_id}/state/solving |Solution object
+|Get best solution a|GET a|solvers/${solver_id}/bestsolution |SolverInstance object including the best solution
+|Terminate solver a|POST a|solvers/${solver_id}/state/terminating-early |-
+|Dispose solver a|DELETE a|solvers/${solver_id} |-
+|===
+
+Apart from REST/JMS API, the server ships with Java client, which provides a common API regardless of the access type used.
+
+== How a typical workflow looks like?
+
+. Install your optimization app kjar to a maven repository accessible by the KIE Server.
+This step assumes the following conditions are met:
+
+* `kmodule.xml` file is located in `META-INF` folder of your project.
+* Project uses packaging type `kjar`.
+* `kie-maven-plugin` is configured for the project.
++
+[source,xml,options="nowrap"]
+----
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.kie</groupId>
+        <artifactId>kie-maven-plugin</artifactId>
+        <version>${version.org.kie}</version>
+        <extensions>true</extensions>
+      </plugin>
+    </plugins>
+  </build>
+----
+
+. Create a container (server unit of execution) wrapping the project.
+This example uses Java KIE Server client, which requires the following dependencies in your client application:
++
+[source,xml,options="nowrap"]
+----
+  <dependency>
+    <groupId>org.kie.server</groupId>
+    <artifactId>kie-server-api</artifactId>
+    <version>7.2.0.Final</version>
```

Comment:
Normally the should bue using the bom, so you can leave of the version (and I wouldn't mention the bom).

Avoiding mentioning a version in copy-paste code is a good thing (because users copy paste the code). It does make sense to mention "the earliest version that this works" somewhere in the proza. Something like "This feature works in 7.2.0.Final and later."